app/core/services/PageBuilder.scala

```diff
@@ -16,15 +16,15 @@
 
 package core.services
 
-import javax.inject.Singleton
+import javax.inject.{Inject, Singleton}
 import core.models.ocelot._
 import core.models.ocelot.stanzas._
 import core.models.ocelot.errors._
 import play.api.Logger
 import scala.annotation.tailrec
 
 @Singleton
-class PageBuilder extends ProcessPopulation {
+class PageBuilder @Inject() (val placeholders: Placeholders) extends ProcessPopulation with PlaceholderProvider {
```

Comment:
Is it necessary to inject an instance of PlaceHolders as it is not being used here.

Reply:
yes, ProcessPopulation expects whatever it is mixed into to be a PlaceholderProvider